### DISCUSSION TITLE
portkey server is not working locally

### DISCUSSION THREAD

**Discussion by jankigabani** [Portkey-AI/gateway#458] [General]
I'm trying to use portkey gateway completely locally. i have passed this cmd and got below

`npx @portkey-ai/gateway
Your AI Gateway is now running on http://localhost:8787 🚀
Returned in Retry Attempt 1. Status: true 200
Returned in Retry Attempt 1. Status: true 200`

in short i'm not getting ui on server : http://localhost:8787/v1

i'm getting this 
`{"status":"failure","message":"Either x-portkey-config or x-portkey-provider header is required"}`

and i have just tried your code present on colab 

`client = OpenAI(
    api_key=OPENAI_API_KEY,
    base_url=PORTKEY_GATEWAY_URL,
    default_headers=createHeaders(
        provider="openai",
        api_key=PORTKEY_API_KEY
    )
)

chat_complete = client.chat.completions.create(
    model="gpt-4",
    messages=[{"role": "user",
               "content": "What's a fractal?"}],
)

print(chat_complete.choices[0].message.content)`

help me to fix this

**Comment by narengogi** [comment]
There is no UI bundled with the gateway @jankigabani 
If you go to the server endpoint http://localhost:8787/ you'll see a message saying "AI Gateway says hey!"

**Comment by jankigabani** [comment]
but wha about i'll go on this server : http://localhost:8787/v1 

still i'm not able to see the anything i'm getting error : `{"status":"failure","message":"Either x-portkey-config or x-portkey-provider header is required"}`

**Reply by narengogi** [reply]
remove the /v1 at the end of the URL @jankigabani

**Comment by jankigabani** [comment]
yes, all i'm saying is that if i remove /v1 then `i'm getting AI Gateway says hey!` but i want to see the dashboard of portkey 

and i followed this file : https://github.com/Portkey-AI/gateway

is it possible to see the dashboard while running locally ? --> like this -> https://portkey.ai/docs/welcome/make-your-first-request

i didn't entered the portkey api key because here in above readme file it is written that : not needed when running locally

`from openai import OpenAI
from portkey_ai import PORTKEY_GATEWAY_URL, createHeaders

gateway = OpenAI(
    api_key="ANTHROPIC_API_KEY",
    base_url=PORTKEY_GATEWAY_URL, # Or http://localhost:8787/v1 when running locally
    default_headers=createHeaders(
        provider="anthropic",
        api_key="PORTKEY_API_KEY" # Grab from https://app.portkey.ai # Not needed when running locally
    )
)

chat_complete = gateway.chat.completions.create(
    model="claude-3-sonnet-20240229",
    messages=[{"role": "user", "content": "What's a fractal?"}],
    max_tokens=512
)`

**Reply by narengogi** [reply]
I don't see what the confusion is @jankigabani, I've already said that there is no UI bundled alongwith the gateway. The dashboard is a part of the hosted version that Portkey provides

**Comment by jankigabani** [comment]
okay, let me ask one question how can i see/get the porkey dashboard on localhost? do you have any resources? @narengogi

**Comment by vrushankportkey** [comment]
Hi @jankigabani - there is currently a difference in the open source v/s hosted version of Portkey. The hosted version is available on `https://app.portkey.ai/` whereas the open source version can be installed by running `npx @portkey-ai/gateway`. 


The open source version is completely headless - it does not come with any sort of UI. It only spins up a Portkey Gateway server on your local system with which you can route your requests and apply fallback, loadbalancing and other mechanisms using the Configs. 

Whereas the hosted version comes in built with observability, and the full dashboard where you can see each request, it's cost/token details, and more.

Please let me know if this clears it up! Happy to help you setup the hosted version if you'd like as well!

**Reply by ZYJ-3721** [reply]
https://portkey.ai/docs/api-reference/admin-api/control-plane/configs/create-config
The code that manages the API is also not use locally?   Localhost: 8787/v1

**Reply by vrushankportkey** [reply]
Hey @AI3721 you're correct. The Gateway does not ship with any of the Admin APIs (like createConfig) that you see here.

**Reply by vrushankportkey** [reply]
However, you can pass raw configs as JSON with the `x-portkey-config` header in any of your requests to your local Gateway. Just that the UI + Configs storage and versioning is not supported on the open source version.

Does this make sense?

**Reply by ZYJ-3721** [reply]
okay, thank you

**Comment by jankigabani** [comment]
okay. thank you (understood)
